### ISSUE TITLE
coverageLocations fails with more than once package

### ISSUE THREAD

**Issue by awentzel** [paambaati/codeclimate-action#153]
Using coverageLocations fails to work.  I've tried many different options and the only one that seems to work is running on a single package.

```bash
 - name: Collect and upload code coverage to Code Climate
      uses: paambaati/codeclimate-action@v2.5.6
      env: 
        CC_TEST_REPORTER_ID: xxxx
      with:
        coverageCommand: yarn run jest --coverage
        coverageLocations:
          "./packages/react/fast-components-foundation-react/coverage/lcov.info:lcov",
          "./packages/react/fast-components-react-base/coverage/lcov.info:lcov"
          # './packages/react/fast-components-react-msft/coverage/lcov.info:lcov',
          # './packages/react/fast-components-styles-msft/coverage/lcov.info:lcov',
          # './packages/react/fast-jss-manager-react/coverage/lcov.info:lcov',
          # './packages/react/fast-jss-utilities/coverage/lcov.info:lcov',
          # './packages/react/fast-layouts-react/coverage/lcov.info:lcov',
          # './packages/react/fast-react-utilities/coverage/lcov.info:lcov',
          # './packages/tooling/fast-tooling/coverage/lcov.info:lcov',
          # './packages/tooling/fast-tooling-react/coverage/lcov.info:lcov',
          # './packages/utilities/fast-animation/coverage/lcov.info:lcov',
          # './packages/utilities/fast-colors/coverage/lcov.info:lcov',
          # './packages/utilities/fast-web-utilities/coverage/lcov.info:lcov'
        debug: true
```

**Comment by paambaati** [comment]
@awentzel Can you link me to a failing run?

**Comment by awentzel** [comment]
Hi @paambaati - here is a draft pr with update `v2.5.7` https://github.com/microsoft/fast-dna/actions/runs/83229740 

Also, did this Action on GH Marketplace get removed?  My bookmark to it is now 404ing. 

Thanks,
Aaron

**Comment by MartinNuc** [comment]
I have the same problem. `coverageLocations` doesn't accept an array like mentioned in docs but just a string.

I believe the reason is described here: https://github.community/t5/GitHub-Actions/Can-action-inputs-be-arrays/td-p/33776

> Action inputs only supports string keys and string values at this time. This is because they translate to environment variables within the action's execution run.

There are two workarounds:

1. use JSON
2. use string with spaces/newlines and split it into the array in the code

I will try it and make a PR 🙂

**Comment by awentzel** [comment]
Thanks @MartinNuc, appreciate your help.  I was going to submit some doc updates as well as soon as I got it figured out.  Were you able to get it to actually format_coverage, sum_coverage, and upload_coverage?  If that works it's a step in the right direction for sure.  The script methods I used on Circle CI seem to be broken on GH Actions now with the cc-test-reporter.  In fact, cc-test-reporter itself doesn't seem to handle mono-repos well.  Thanks again.

**Comment by MartinNuc** [comment]
Not yet. Still working on it. I had to convert also prefix into array. So files are being passed now as a multiline string (last segment after `:`):

```
        with:
          coverageLocations: |
            ${{github.workspace}}/client/coverage/lcov.info:lcov:${{github.workspace}}/client
            ${{github.workspace}}/server/coverage/lcov.info:lcov:${{github.workspace}}/server
```

Right now have a problem that `format_coverage` is trying to read files without prefix. I will have to try it locally first and then I will submit PR.

**Comment by awentzel** [comment]
@MartinNuc , I did a test and got through that part thanks to your suggestion.  But, haven't had any luck yet getting format_coverage to work.  In fact, I have a few threads on this issue, including a different approach scripting in the CI rather than using an Action.  But, all roads for that have resulted in `no such file or directory`.  Very challenging problem on mono-repositories.  What `coverageCommand` are you using?  Technically, unit testing has already created my `/coverage` data.  But, it's unclear how things process inside this black box.  I haven't dug into the source code yet.

**Comment by MartinNuc** [comment]
Finally got it working. My problem was actually with the instanbul coverage file. There were relative urls which didn't work well with monorepo. I had to set the `projectRoot` in jest config.

Here is the issue if someone had the same problem: https://github.com/istanbuljs/istanbuljs/issues/529

It turned out that there is no need to use different prefix for each covarageLocation therefore I removed that feature.

If you want you can try my version using:

```
      - name: Coverage to Codeclimate
        uses: MartinNuc/codeclimate-action@v2.5.10
        env:
          CC_TEST_REPORTER_ID: ${{ secrets.CODECLIMATE_REPORTER_ID }}
        with:
          coverageLocations: |
            ${{github.workspace}}/client/coverage/lcov.info:lcov
            ${{github.workspace}}/server/coverage/lcov.info:lcov
```

I will post the PR.

**Comment by paambaati** [comment]
@MartinNuc Would be super-helpful if you can add these to the README's examples section as well.

**Comment by MartinNuc** [comment]
@paambaati Actually I did here https://github.com/paambaati/codeclimate-action/pull/163/files#diff-04c6e90faac2675aa89e2176d2eec7d8L44

**Comment by paambaati** [comment]
Thanks a ton @MartinNuc! This fix has now landed in [`v2.6.0`](https://github.com/paambaati/codeclimate-action/releases/tag/v2.6.0).
